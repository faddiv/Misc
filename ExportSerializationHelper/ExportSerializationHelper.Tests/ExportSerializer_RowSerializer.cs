using CsvHelper;
using CsvHelper.Configuration;
using ExportSerializationHelper.Tests.Data;
using FluentAssertions;
using Snapshooter.Xunit;
using System.Globalization;
using System.IO;
using System.Linq;
using Xunit;

namespace ExportSerializationHelper.Tests;

public class ExportSerializer_RowSerializer
{
    [Fact]
    public void RowSerializer_Exports_Into_Csv()
    {
        var config = new ExampleClassMap();
        var data = TestData.GenerateExampleClasses();
        using var stream = new MemoryStream();
        using var writer = new StreamWriter(stream);
        using var csv = new CsvWriter(writer, new CsvConfiguration(CultureInfo.InvariantCulture)
        {
            HasHeaderRecord = false
        });

        foreach (var header in config.GetHeader())
        {
            csv.WriteField(header);
        }
        csv.NextRecord();

        foreach (var item in ExportSerializer.RowSerializer(data, config))
        {
            foreach (var field in item.Data)
            {
                csv.WriteField(field);
            }
            csv.NextRecord();
        }
        csv.Flush();
        var result = writer.Encoding.GetString(stream.ToArray());

        result.Should().MatchSnapshot();
    }

    [Fact]
    public void RowSerializer_Exports_Into_ClosedXml()
    {
        var config = new ExampleClassMap();
        var data = TestData.GenerateExampleClasses();
        using var stream = new MemoryStream();
        var workbook = new ClosedXML.Excel.XLWorkbook();
        var worksheet = workbook.AddWorksheet("Sheet1");
        worksheet.Cell(1, 1).InsertData(new[] { config.GetHeader() });
        worksheet.Cell(2, 1).InsertData(ExportSerializer.RowSerializer(data, config).Select(e => e.Data));

        ResultData.TextExcel(worksheet);
    }

}
