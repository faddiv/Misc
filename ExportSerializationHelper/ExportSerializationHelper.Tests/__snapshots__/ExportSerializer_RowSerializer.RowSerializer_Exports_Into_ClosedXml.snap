[
  {
    "Identifier": 399.0,
    "Some Text": "wbycytinbf",
    "Timestamp": "2022-03-05T03:08:43.579",
    "Nullable": ""
  },
  {
    "Identifier": 464.0,
    "Some Text": "qol224slj4",
    "Timestamp": "2022-03-05T08:39:51.055",
    "Nullable": "2022-03-05T02:47:03.784"
  },
  {
    "Identifier": 284.0,
    "Some Text": "ewrhixj87c",
    "Timestamp": "2022-03-05T07:58:05.016",
    "Nullable": ""
  },
  {
    "Identifier": 842.0,
    "Some Text": "j5xii6spbr",
    "Timestamp": "2022-03-05T02:48:21.26",
    "Nullable": "2022-03-05T02:42:45.146"
  },
  {
    "Identifier": 278.0,
    "Some Text": "sc6b42i4qk",
    "Timestamp": "2022-03-05T01:55:09.227",
    "Nullable": "2022-03-05T23:28:56.134"
  },
  {
    "Identifier": 936.0,
    "Some Text": "ih8ega8gml",
    "Timestamp": "2022-03-05T20:46:15.27",
    "Nullable": "2022-03-05T21:43:06.734"
  },
  {
    "Identifier": 583.0,
    "Some Text": "24v2mnqw01",
    "Timestamp": "2022-03-05T10:58:32.402",
    "Nullable": ""
  },
  {
    "Identifier": 318.0,
    "Some Text": "fecl55u7uj",
    "Timestamp": "2022-03-05T12:33:44.516",
    "Nullable": ""
  },
  {
    "Identifier": 604.0,
    "Some Text": "1cql62yx1m",
    "Timestamp": "2022-03-05T07:25:20.867",
    "Nullable": ""
  },
  {
    "Identifier": 720.0,
    "Some Text": "d7mg6vwys1",
    "Timestamp": "2022-03-05T17:33:19.061",
    "Nullable": "2022-03-05T00:27:30.918"
  }
]
